app/components/chat/UserMessage.tsx

```diff
@@ -9,13 +9,34 @@ interface UserMessageProps {
 }
 
 export function UserMessage({ content }: UserMessageProps) {
+  const sanitizedContent = sanitizeUserMessage(content);
+  const textContent = Array.isArray(sanitizedContent) 
+    ? sanitizedContent.find(item => item.type === 'text')?.text || ''
+    : sanitizedContent;
```

Comment:
can this be merged/inserted inside `sanitizeUserMessage` function? otherwise you are unnecessarily looping twice over the contents